### PR TITLE
[kubectl] New app

### LINKED ISSUE #111: [kubectl] create new image
# Details

**Image name:**

kubectl

**Describe the solution you'd like:**

A container image with kubectl, jq and curl versioned with the kubectl version tag

### PR DESCRIPTION
**Pull request by nicholaswilde**
Signed-off-by: Nicholas Wilde <ncwilde43@gmail.com>

**Description of the change**

Adds kubectl with jq and curl.

**Benefits**

**Possible drawbacks**

**Applicable issues**

- fixes #111 

**Additional information**

`jq` and `curl` are installed via `apt`

`CMD` is used rather than `ENTRYPOINT` and so you can easily call `jq` or `curl`

```
# kubectl
docker run --rm kubectl:latest
# jq
docker run --rm kubectl:latest jq
# curl
docker run --rm kubectl:latest curl
```

### CODE REVIEW

File: apps/kubectl/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+# hadolint ignore=DL3007
+FROM ghcr.io/k8s-at-home/ubuntu:latest
+WORKDIR /app
+USER root
+# hadolint ignore=DL3008
+RUN \
+  apt-get -qq update \
+  && apt-get -qq install -y --no-install-recommends \
+    jq \
+    curl \
```
Comment (onedr0p) [inline apps/kubectl/Dockerfile:26]: We actually have curl in the base image :)

File: apps/kubectl/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+RUN \
+  apt-get -qq update \
+  && apt-get -qq install -y --no-install-recommends \
+    wget \
+  && wget -nv "https://dl.k8s.io/release/v${VERSION}/bin/linux/${TARGETARCH}/kubectl" \
+  && chmod +x kubectl
+
+# hadolint ignore=DL3007
+FROM ghcr.io/k8s-at-home/ubuntu:latest
+WORKDIR /app
```
Comment (onedr0p) [inline apps/kubectl/Dockerfile:19]: This is set in the base image

File: apps/kubectl/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+USER root
+# hadolint ignore=DL3008
+RUN \
+  apt-get -qq update \
+  && apt-get -qq install -y --no-install-recommends \
+    jq \
+    curl \
+  && chown -R kah:kah /app \
```
Comment (onedr0p) [inline apps/kubectl/Dockerfile:27]:
```suggestion
  && \
  curl -fsSL -o /app/kubectl "https://dl.k8s.io/release/v${VERSION}/bin/linux/${TARGETARCH}/kubectl"
  && chown -R kah:kah /app \
```

File: apps/kubectl/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+ARG VERSION
+WORKDIR /tmp
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+# hadolint ignore=DL3008
+RUN \
+  apt-get -qq update \
+  && apt-get -qq install -y --no-install-recommends \
+    wget \
+  && wget -nv "https://dl.k8s.io/release/v${VERSION}/bin/linux/${TARGETARCH}/kubectl" \
+  && chmod +x kubectl
```
Comment (onedr0p) [inline apps/kubectl/Dockerfile:15]: Don't really need a multi-stage build for this.

File: apps/kubectl/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+# hadolint ignore=DL3007
+FROM ghcr.io/k8s-at-home/ubuntu:latest
+WORKDIR /app
+USER root
```
Comment (onedr0p) [inline apps/kubectl/Dockerfile:20]:
```suggestion
ARG TARGETARCH
ARG VERSION

USER root
```

**Review by onedr0p — CHANGES_REQUESTED** [review]
Only a few changes I think we can do.

**Comment by nicholaswilde** [comment]
> Only a few changes I think we can do.

I guess it helps to pay attention to the base image while doing development. 😝 Thanks for the review.

File: apps/kubectl/Dockerfile (hunk 5 above)
Comment (nicholaswilde) [reply apps/kubectl/Dockerfile]: Kept root at single stage in order to install jq.

File: apps/kubectl/Dockerfile (hunk 4 above)
Comment (nicholaswilde) [reply apps/kubectl/Dockerfile]: Converted to single stage.

File: apps/kubectl/Dockerfile (hunk 2 above)
Comment (nicholaswilde) [reply apps/kubectl/Dockerfile]: Done.

File: apps/kubectl/Dockerfile (hunk 1 above)
Comment (nicholaswilde) [reply apps/kubectl/Dockerfile]: Removed.

**Review by onedr0p — APPROVED** [review]

**Pull request merged**